File: leetcode/lc-0058/solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        bool in_word = false;
        int length = 0;
        for (auto it = s.rbegin(); it != s.rend(); it++) {
            if (*it == ' ') {
                if (in_word) break;
            } else {
                in_word = true;
                length++;
            }
        }
        return length;
    }
};

File: leetcode/lc-0058/solution.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let mut in_word = false;
        let mut len = 0;
        for ch in s.chars().rev() {
            match ch {
                ' ' => {
                    if in_word {
                        break;
                    }
                },
                _ => {
                    in_word = true;
                    len += 1;
                }
            }
        }
        len
    }
}
